### ISSUE TITLE
Bug: Runtime crash when handling string with only 1 character

### ISSUE THREAD

**Issue by timofticiuc** [Smartling/ios-lqa-sdk#10]
First of all: I've only observed this once and was able to reproduce, now to have it not happen at all

Context: 
I have a string that had been previously localized (NSLocalizedString("%")) and smartling was setup to use SLContextCapture.
During app launch, right after Smartling.start, i hit this exception:

```
-[NSTaggedPointerString FBsubstringWithRange:]: Range {0, 2} out of bounds; string length 1
(
0   CoreFoundation                      0x0000000114e5b1e6 __exceptionPreprocess + 294
1   libobjc.A.dylib                     0x0000000112f06031 objc_exception_throw + 48
2   CoreFoundation                      0x0000000114ed0975 +[NSException raise:format:] + 197
3   CoreFoundation                      0x0000000114dad9ff -[NSTaggedPointerString substringWithRange:] + 591
4   DebugApp                          0x000000010ab79fb7 -[NSString(Smartling) sl_stringWithCues] + 504
5   DebugApp                            0x000000010ab7bd86 +[SLBaseLocaleStrings loadStringsFromLocalizableStringsAtPath:intoDict:] + 1179
6   DebugApp                            0x000000010ab7b89a +[SLBaseLocaleStrings localeStringsForLocale:] + 293
7   DebugApp                            0x000000010ab73cfe -[SLMainController setupBaseLocale] + 187
8   DebugApp                            0x000000010ab71ee0 -[SLMainController startWithProjectId:andOptions:] + 1224
9  DebugApp                            0x000000010ab71186 +[Smartling startWithProjectId:andOptions:] + 97
```

i swizzled into substringWIthRange so as to get a clue into what's causing the substring to go out of bounds
```
@implementation NSString (Timo)

+ (void)load {
	method_exchangeImplementations(class_getInstanceMethod(NSClassFromString(@"NSTaggedPointerString"), @selector(substringWithRange:)),
								   class_getInstanceMethod(NSClassFromString(@"NSTaggedPointerString"), @selector(FBsubstringWithRange:)));
}

- (NSString *)FBsubstringWithRange:(NSRange)range {
	return [self FBsubstringWithRange];
}

@end
```

This yielded the string to be "%", over which a substringWithRange was being applied with a location of 0 and a length of 2
![screen shot 2019-01-09 at 00 53 31](https://user-images.githubusercontent.com/9297289/50906322-83dea080-142d-11e9-8d65-7cc17d644421.png)

I assume this by how smartling works under the hood, presumably swizzling into NSLocalizedString?
After commenting removing the usage of "%", the runtime crash never occurred and i haven't been able to reproduce since.
Are one-char strings an issue? for example i would like to localize "I" which would translate to "Ich" in german

Thanks in advance,
timo

**Comment by enegron-sl** [comment]
We can investigate this as a bug, but I will also suggest that you are not following localization best practices.  If you are translating something like "%" or "I" by itself then you are probably concatenating strings, and that is not advisable since the translation of specific words will depend on their context and if you translate and then concatenate words (or characters, or phrases) then you probably will end up with bad translations.  I suggest you form complete phrases or sentences in your strings.  

The main use case for single word strings are UI elements (labels, buttons, etc).  But even then characters, words or phrases shouldn't be concatenated.

So instead "%" write a complete string, "%d% complete"  and instead of "I" write the phrases you need completely, for example: "I am available", "I am not available"

**Comment by timofticiuc** [comment]
Agreed with the incorrect use case of "%". It used to be a longer string that was localized, but then kept being split up till only "%" remained. For now i've left it as unlocalized since it doesn't make sense to localize a symbol which indeed breaks localization best practices :)

However, i'd like to reiterate that right now the framework is unusable since i'll have cases where there will be the need to localize a single char string. The business logic should be agnostic of how long a localized string is

**Comment by edwardvalentini** [comment]
It's a serious issue for the API to crash for 1 character strings.... I can think of several uses for 1 character labels... The api shouldn't crash.... it should handle 1 and even 0 length strings gracefully.

**Comment by mbarthelemy** [comment]
Published API's should not crash on index-out-of-range errors regardless of the input. I will have Fitbit's internationalization team raise this issue with our contacts at Smartling. This crash is blocking our integration of the Framework.

**Comment by enegron-sl** [comment]
we've confirmed the issue; it's not the character count but "%" specifically.

**Comment by timofticiuc** [comment]
@enegron-sl can we get an ETA or a ping when this is fixed? thanks :)

**Comment by timofticiuc** [comment]
@enegron-sl this is a blocker for us integrating the framework. Can you advise when this will get worked on? We will escalate this through our business contacts.

**Comment by foo4u** [comment]
Fixed in v1.5.2
